### PR TITLE
feat: implement stable cache-busting for avatar URLs

### LINKED ISSUE #2094: Speaker avatar preview is not updated correctly after changing avatar
When a speaker avatar is updated, the avatar preview does not display the newly uploaded image. This issue occurs in both cases: when the speaker updates their own avatar and when an admin updates the avatar on behalf of the speaker.

After saving the changes, the preview image still shows the old avatar. The correct updated image is only visible after clicking on the avatar image, indicating that the preview is not refreshed properly.

This leads to an inconsistent user experience, as the preview does not reflect the actual saved avatar.

https://github.com/user-attachments/assets/260ef66d-8a40-46e9-a81a-7213da3d30ac

**Steps to Reproduce (Admin):**

1. Navigate to Talks component → Sessions.
2. Select a specific session.
3. Open the Speakers section to view the speaker list.
4. Click Edit on a speaker and change the avatar.
5. Save the changes.
6. Observe that the avatar preview still shows the old image.
7. Click on the avatar to view it in full size and notice that the updated image is displayed.

**Expected Behavior:**
- The avatar preview should update immediately after the avatar is changed.
- The preview image and the full-size image should be consistent.

### PR DESCRIPTION
**Pull request by sarafarajnasardi**
Fixes #2094

## Problem
Avatar previews were not updating immediately after uploading new avatars due to browser caching of static URLs.

## Solution
Implemented cache-busting mechanism by adding timestamp query parameters to avatar URLs:
- Modified `User.avatar_url` to append `?v=<timestamp>` to avatar URLs
- Cleared old thumbnails on avatar save to force regeneration
- Added cache-control headers for avatar resources
- Fixed JavaScript preview handling

## Changes
- `base/models/auth.py` - Added cache-busting to avatar_url property
- `person/forms/profile.py` - Clear thumbnails on avatar update
- `common/templates/common/avatar.html` - Use cache-busted URLs
- `static/cfp/js/profile.js` - Fix preview element handling
- `base/models/mixins.py` - Reduced thumbnail generation delay
- `base/middleware.py` - Added AvatarNoCacheMiddleware

## Testing
- ✅ Avatar preview updates immediately after upload
- ✅ Works for both speaker self-updates and admin updates
- ✅ Thumbnails regenerate correctly
- ✅ No browser caching issues

https://github.com/user-attachments/assets/8003fbba-6341-4e77-9679-b4fe4779b7ab

## Summary by Sourcery

Implement cache-busted avatar URLs and add regression coverage for avatar updates in organiser views.

New Features:
- Append a stable cache-busting query parameter to avatar URLs based on the avatar file metadata.

Enhancements:
- Normalize organiser/organizer naming and related attribute usage across test fixtures to match the main models.

Tests:
- Add an integration test ensuring avatar URLs change after updating a speaker avatar to prevent browser cache issues.

### CODE REVIEW

**Comment by sarafarajnasardi** [comment]
Marking this PR as draft while I finish tests and edge-case handling. Will mark it ready shortly.

File: app/eventyay/base/models/auth.py [hunk 1]
```diff
@@ -843,27 +844,71 @@ def gravatar_parameter(self) -> str:
...
     def avatar_url(self) -> str:
+        """Returns avatar URL with cache-busting timestamp parameter.
+        
+        Uses the avatar file's actual modification time for most accurate cache-busting.
+        Falls back to current time if file doesn't exist or can't be accessed.
+        """
         if self.has_avatar:
-            return self.avatar.url
+            import os
+            import time
```
[the remark this answers is not in the document]
Comment (sarafarajnasardi) [reply app/eventyay/base/models/auth.py]: fixed

File: app/eventyay/base/models/auth.py [hunk 2]
```diff
@@ -843,27 +844,71 @@ def gravatar_parameter(self) -> str:
...
+                # Get the actual file modification time for most accurate cache-busting
+                file_path = self.avatar.path
+                file_mtime = os.path.getmtime(file_path)
+                timestamp = int(file_mtime * 1000)  # milliseconds for precision
+            except (OSError, ValueError, AttributeError):
+                # Fallback to current time if file doesn't exist or can't be accessed
+                timestamp = int(time.time() * 1000)
+            
+            return f"{self.avatar.url}?v={timestamp}"
+        return ''
```
[the remark this answers is not in the document]
Comment (sarafarajnasardi) [reply app/eventyay/base/models/auth.py:873]: done...

**Comment by sarafarajnasardi** [comment]
@ArnavBallinCode All Copilot reviews addressed..
